Work toward automating release management.

jmchilton opened the pull request:
Create new `Makefile` targets for release management.
- `release-create-issue`: Automates creating a release checklist - [check out example](https://github.com/jmchilton/galaxy/issues/41).
  
  Issue is nested check list containing actual tailored `make` commands and links (for PRs and issues).
- `release-check-metadata`: Checks milestone PRs for required metadata needed for generating bootstrapped release notes.
- `release-bootstrap-history`: Bootstrap release notes from PRs and metadata.
- `release-check-blocking-prs`: Check for PRs blocking release, should happen before branching release and before tagging release.
- `release-check-blocking-issues`: Check for Issues blocking release, should happen before tagging release. (Only allowed issue is the release creation issue.)

The issue created with `release-create-issue` documents how and when to execute subsequent `make` targets (including the target to create the issue for the next release).

This PR contains a bunch of other enhancements to the bootstrap_history.py script to enable these targets and renames `create_release_rc` Makefile target so all release related targets start with the prefix `release-`. This PR also adds some more Makefile targets used by planemo, pulsar, galaxy-lib, etc... that have proved useful to me in the past.

jmchilton commented:
- @natefoo Can you update the source of the release issue generator as you do the release for 16.01?
  
  I suspect 16.04 will still only be "releasable" ("Do Release" section) by @natefoo - but we should aim to have someone else do 16.07 (8?) at least.
- @jennaj Can you review the check list https://github.com/jmchilton/galaxy/issues/41 and jot down notes to expand it and provide details as you announce 16.01? Specific details such as accounts and how to find passwords in password store would be great. The goal would be for someone else to be able to announce 16.04 based on your notes. I volunteer to do that. Whoever does 16.04 can then fill in details that seem obvious to someone that has done this before.
- I think the release automation and documentation in-issue are good enough for someone else to take over release note generation for 16.04 - any volunteers? (I'm trying to reduce my own bus factor also :grinning:.)
- @natefoo Based on your existing notes and the example issue, do you think someone else could do "Branch Release" for 16.04? If so, I'd volunteer.
- @natefoo "Deploy and Test Release" is a bit sparse currently - do you think you can fill in some details and pass that off for 16.04 to someone else @martenson or @davebx for instance?

natefoo commented:
> @natefoo Can you update the source of the release issue generator as you do the release for 16.01?

Yep.

> @natefoo Based on your existing notes and the example issue, do you think someone else could do "Branch Release" for 16.04? If so, I'd volunteer.

Yep.

> @natefoo "Deploy and Test Release" is a bit sparse currently - do you think you can fill in some details and pass that off for 16.04 to someone else @martenson or @davebx for instance?

Yep.

jennaj commented:
@jennaj Can you review the check list jmchilton#41 and jot down notes to expand it and provide details as you announce 16.01? Specific details such as accounts and how to find passwords in password store would be great. The goal would be for someone else to be able to announce 16.04 based on your notes. I volunteer to do that. Whoever does 16.04 can then fill in details that seem obvious to someone that has done this before.

Will do

jmchilton commented:
These test failures have nothing to do with the PR. A fix for the transiently failing API test has been merged and the toolshed tests are known to be transiently failing.

jmchilton committed:
Work toward automating deployment.

Create new makefile targets for release management.

 - ``release-create-issue``: Automates creating a release checklist - https://github.com/jmchilton/galaxy/issues/41)
   Check list contains actual tailored ``make`` commands and links (for PRs and issues).
 - ``release-check-metadata``: Checks milestone PRs for required metadata needed for generating bootstrapped release notes.
 ` ``release-bootstrap-history``: Bootstrap release notes from PRs and metadata.
 - ``release-check-blocking-prs``: Check for PRs blocking release, should happen before branching release and before tagging release.
 - ``release-check-blocking-issues``: Check for Issues blocking release, should happen before tagging release. (Only allowed issue is the release creation issue.)

The issue created with ``release-create-issue`` documents how and when to execute subsequent ``make`` targets (including the target to create the issue for the next release).

This PR contains a bunch of other enhancements to the bootstrap_history.py script to enable these targets and renames ``create_release_rc`` Makefile target so all release related targets start with the prefix ``release-``.

Conflicts:
	Makefile

martenson commented:
thanks @jmchilton 

I am volunteering to do "Deploy and Test Release" for 16.04 (@natefoo please help with 'filling the missing details')

martenson commented:
xref https://github.com/galaxyproject/galaxy/issues/1979

martenson commented:
I am hereby de-volunteering for deployment and volunteering for the release notes work. I will help with the deployment if needed though :)
